File: 3351-maximize-happiness-of-selected-children/3351-maximize-happiness-of-selected-children.java
class Solution {
    public long maximumHappinessSum(int[] happiness, int k) {
        Arrays.sort(happiness);
        long ans = 0;

        for (int i = happiness.length - 1; i >= 0 && k-- > 0; i--) {
            int val = (happiness[i] - (happiness.length - i - 1));
            if (val > 0) {
                ans += val;
            }
        }

        return ans;
    }
}

File: 3351-maximize-happiness-of-selected-children/3351-maximize-happiness-of-selected-children.py
class Solution:
    def maximumHappinessSum(self, happiness: List[int], k: int) -> int:
        happiness.sort(reverse=True)
        ans = 0
        maxrange = min(len(happiness), k)
        for i in range(maxrange):
            val = happiness[i] - i
            ans += val if val > 0 else 0
        return ans
        